Fix field name styling in `submitpackage` RPC results

tnull opened the pull request:
Previously, the `submitpackage` RPC call returned a response key named `package_msg`, which is inconsistent with the other field names that are using `-` as a delimiter. Here, we align the style of `package-msg`.

cc @glozow @instagibbs

glozow reviewed (commented):
lgtm, though I've heard we don't like hyphens, so perhaps go in the other direction and change all the other ones to `_`?

glozow commented:
I want to say this changed over time... I can't find the thread where I was convinced, but here's chat GPT's argument for why underscores are better than hyphens in JSON:

1. Consistency with JavaScript & JSON standards – JSON keys are commonly accessed in JavaScript, and using hyphens (some-key) requires bracket notation (obj["some-key"]) instead of the simpler dot notation (obj.some_key).
2. Better compatibility – Many programming languages (Python, JavaScript, etc.) use underscores for variable and property names, making underscore-separated keys more natural to work with.
3. Avoids potential parsing issues – Some parsers and serialization libraries may misinterpret hyphens as minus signs, leading to unexpected issues."

fwiw, if we think of pre-29 as the last chance to be liberal with changing API, I do think unifying it within `submitpackage` makes sense (for context, this is a continuation of the convo from https://github.com/spesmilo/electrum-protocol/issues/4#issuecomment-2665116596).

stickies-v reviewed (commented):
The developer notes [state](https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#rpc-interface-guidelines):

> Argument and field naming: please consider whether there is already a naming style or spelling convention in the API for the type of object in question (blockhash, for example), and if so, try to use that. If not, use snake case fee_delta (and not, e.g. feedelta or camel case feeDelta).
> 
> Rationale: Consistency with the existing interface.

So it seems like if any change is made, it should be `package_msg`, or update developer notes first. The JSON-RPC spec does not provide guidelines on using either or the other, and it doesn't seem to me like there is an industry standard either, so I don't see a particular reason to update the developer notes.

> fwiw, if we think of pre-29 as the last chance to be liberal with changing API, I do think unifying it within `submitpackage` makes sense

I agree with that. Concept ACK.

tnull committed:
Align `submitpackage` result field names to use snake_case

Previously, the result of a `submitpackage` RPC call would include
fields with mixed styles (mixing dash-case and snake_case). Here, we
align them all to use snake_case, i.e., to use the `_` delimiter.

tnull committed:
Align `testmempoolaccept` result field names to use snake_case

Previously, the result of a `testmempoolaccept` RPC call would include
fields with mixed styles (mixing dash-case and snake_case). Here, we
align them all to use snake_case, i.e., to use the `_` delimiter.

stickies-v commented:
`submitpackage` is marked as experimental and unstable, so changing that is fine imo. That does not hold for `testmempoolaccept`, so if we make changes to fields you'll have to add a `-deprecatedrpc` option. I don't think breaking a stable API just for stylistic purposes is worth it, so I would postpone that until we need another breaking change.

Would be good to add release notes to document the `submitpackage` breaking changes.
